src/lua/rust_interop.rs

```diff
@@ -0,0 +1,174 @@
+//! Rust code which is called from lua in the init file
+#![deny(unused_code)]
+
+use std::sync::Arc;
+use std::ops::Deref;
+
+use hlua::{self, Lua, LuaTable};
+use hlua::any::AnyLuaValue;
+use hlua::any::AnyLuaValue::*;
+use rustc_serialize::json::Json;
+
+use registry::{self, RegistryField, RegistryError, AccessFlags};
+use commands;
+use keys::{self, KeyPress, KeyEvent};
+use convert::json::{json_to_lua, lua_to_json};
+use convert::serialize::ToTable;
+
+type OkayResult = Result<(), &'static str>;
```

Comment:
You're inconsistent with your usage of this Result, and it's generally not really worth it. It might save on a few keypresses, but reading the code it requires a lookup for a very simple type.